channeldb/channel.go

```diff
@@ -2422,9 +2422,7 @@ func (c *OpenChannel) AdvanceCommitChainTail(fwdPkg *FwdPkg,
 
 		// With the commitment pointer swapped, we can now add the
 		// revoked (prior) state to the revocation log.
-		//
-		// TODO(roasbeef): store less
-		err = appendChannelLogEntry(logBucket, &c.RemoteCommitment)
+		err = appendRevocationLogEntry(logBucket, &c.RemoteCommitment)
```

Comment:
AFAICT, this would now require a migration, that has the potential to end up touching millions of keys for the larger existing nodes out there. 

One way to avoid this would be to utilize the fact that the first entry in the serialized channel commitment is actually the commitment height. We use a `unint64` value here as is, so we could use an "extreme" (max uint64) as a shibboleth value to signal that we're actually encoding a different value here. This would work, but is pretty hacky...

Alternatively, we could just use a _new_ key prefix here instead. So the old entries are queried as normal, with the new entries appending/prepending an extra byte. The new logic then inherits non of the prior baggage is it just writers the key/value as normal. When looking up the relevant state number, we need to check the old _and_ the new key, which is an extra check, but in the grand scheme shouldn't lead to any performance issues, as breaches are pretty rare in practice anyway.

Reply:
Yeah I think the second approach is much better, will implement it!